### DISCUSSION TITLE
Welcome to Projects Discussions!

### DISCUSSION THREAD

**Discussion by gabonator** [gabonator/Projects#5] [General]
## 👋 Welcome!
  If you have any questions regarding these project, you can ask here for more details

**Comment by thedabezt** [comment]
Hi @gabonator !! I hope I find you doing well!
I've tried to run ali.js with Node.js in my ubuntu laptop (but it does not work) in order to 'discover' my Viark 4k stb box / decoder, to get the stream urls, etc. 
I think it uses the same alifen method (rtsp://ip:554/stream=X) because it can be streamed to GMScreen android app, but I would like to stream it to VLC or something like that...
 I do not get any answer when run "node ali.js", no result, no answer, no end.... 
 Of course I've changed 192.168.1.50 box ip to my real one ... (if ran with no right IP, i got a connection error)
 Thanks!!!

**Reply by gabonator** [reply]
How much are you familiar with javascript? try adding some debugging outputs on connection and on data handlers, maybe it could give us some clue what is going on:

Change following code:
```
 Ali.socket.connect(port, url, () =>
    {
      Ali.socket.write(Ali._alibuffer("<?xml version='1.0' encoding='UTF-8' standalone='yes' ?><Command request=\"998\" />"));
      Ali.socket.on('data', (data) => Ali.ondata(data));
      Ali.socket.on('close', () => console.log('Connection closed'));
```

to:

```
 Ali.socket.connect(port, url, () =>
    {
      console.log("== on connect");
      Ali.socket.write(Ali._alibuffer("<?xml version='1.0' encoding='UTF-8' standalone='yes' ?><Command request=\"998\" />"));
      Ali.socket.on('data', (data) => {  console.log("== on data", data); Ali.ondata(data) });
      Ali.socket.on('close', () => console.log('Connection closed'));
```

**Reply by thedabezt** [reply]
Thanks  a lot my friend. 
I know not so much about .js in command line with node.js ... I know more about for web front end html embeded javascript coding :-)

But I'm figuring it out to run it and I think I'm progressing. 

I've been able to connect to the stb box and I got a setup url with all the params your code builds up.
Then I see lot of PLAY states with an url in OPTION, like rstp://mystbboxip:554/stream=1

while running your code (refreshing with up + rtsp SEQ states messages, I tried to copy and paste "rstp://mystbboxip:554/stream=1" to VLC and it cannot play it (it stops)
If I copy and paste the SETUP url with all the params it gives me an error

What I've to do in VLC with your code running in a console in order to 'play' the stream?

Thnx!

**Reply by gabonator** [reply]
In vlc, open: "rtp://@:5114"

**Reply by thedabezt** [reply]
Thanks my friend! But nothing is streamed ...¿?

1. It takes many time to start really running ali.js... 1-3 minutes. I've left only one channel in stb box channel list in order to test if it was the 'loading' of lot of channels I originally have. But it takes the same time 1 channel or 2000 :-)

2. When it really starts, it refreshes with up up messages, with same session id, etc. And then in VLC -> ctrl+N -> rtp://@:5114, NO error, it seems to keep trying to get the stream.... VLC keeps showing nothing,and "orange load bar" in the bottom keeps moving side to side, but nothing shown ¿?

I've checked that there's nothing running & listening in port 5114-5115 in my ubuntu laptop where I'm running ali.js, even your code outputs to me:

"Transport: RTP/AVP;unicast;source=Mysetboxip;client_port=5114-5115;server_port=5004-5005;ssrc=A3730158"

Is it supposed to have a LISTEN processes into port 5114 or 5115? in the computer I'm running ali.js? isn't it?

Or it is sent to network broadcast??? should I be able to open from another computer in the network the "rtp://@:5114" stream? I'm not :-(

Thnx again! cheers!

**Reply by thedabezt** [reply]
¿? would you be kind to help me? thanks a lot!
